question17/main.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const imgListsWrapper = document.getElementById('js-imgListsWrapper');
+const imgLists = document.getElementById('js-imgLists');
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'loading';
+  imgListsWrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => {
+  document.getElementById('loading').remove();
+} 
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/09593e78-80ba-4765-940e-3c70fb46078c'));
+  }, 3000);
+})
+
+//Get json data
+async function callApi() {
+  try {
+    const res = await getData;
+    if (!res.ok) {
+      throw new Error(`サーバーリクエストに失敗しました: ${res.status}`);
+    }
+    const json = await res.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+async function init() {
+  const imgData = await callApi();
+  if (imgData){
+    renderImgUiElement(imgData);
+    } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+
+function renderImgUiElement(imgData) {
+  renderNextBtn(imgData);
+  renderPrevBtn(imgData);
+  renderPageNumElement(imgData);
+  imgLists.appendChild(getImgListsFragment(imgData));
+}
```

Comment:
Q.[This is the same question as above.](https://github.com/csakyo/homework/pull/31/files#r835882436)

Reply:
I fixed here too. Please confirm.
fixed commit: https://github.com/csakyo/homework/commit/426f23250e813bfb10ae71c67a386b3ecaa34f10